lib/sles4sap_publiccloud.pm

```diff
@@ -78,9 +91,145 @@ sub get_promoted_hostname {
     return join("", @master);
 }
 
+=head2 get_hana_topology
+    get_hana_topology([hostname => $hostname]);
+
+    Parses  command output, returns list of hashes containing values for each host.
+    If hostname defined, returns hash with values only for host specified.
+=cut
+
+sub get_hana_topology {
+    my ($self, %args) = @_;
+    my @topology;
+    my $hostname = $args{hostname};
+    my $cmd_out = $self->run_cmd(cmd => "SAPHanaSR-showAttr --format=script", quiet => 1);
```

Comment:
Where is it executed?

Reply:
You mean the sub?

Reply:
Ah you mean on which host?
It is executed on whatever is currently defined in "$self->{my_instance}" which is one of the hosts defined in "@instances". This is how all public cloud related modules work in general. 
Hmm... but this is a good point. I will create a check for modules requiring it which will throw error with information how to define it.